tests/virt_autotest/update_package.pm

```diff
@@ -61,9 +61,16 @@ sub run {
     #workaround of bsc#1177790
     #disable DNSSEC validation as it is turned on by default but the forwarders donnot support it, refer to bsc#1177790
     if (is_sle('>=12-sp5')) {
-        script_run "sed -i 's/#dnssec-validation auto;/dnssec-validation no;/g' /etc/named.conf";
-        script_run "grep 'dnssec-validation' /etc/named.conf";
-        script_run "systemctl restart named";
+        #ues die_on_timeout=> 0 as workaround for s390x test during call script_run, refer to poo#106765
+        my %args = ();
+        if (is_s390x) {
+            $args{die_on_timeout} = 0;
+        } else {
+            $args{die_on_timeout} = 1;
+        }
+        script_run "sed -i 's/#dnssec-validation auto;/dnssec-validation no;/g' /etc/named.conf", %args;
+        script_run "grep 'dnssec-validation' /etc/named.conf", %args;
+        script_run "systemctl restart named", %args;
```

Comment:
Similar here, shall s390 and non-s390 both use die => 0?

Reply:
@guoxuguang  I think you'd better raise this issue to openQA project or let them know. It seems that the change does not take this usage on s390x into consideration which means you have to use die_on_timeout => 0. Now you made the change which branches off s390x. I did not say branching off here is really bad here, but it does force us to make the same change to other arches here to some extent because it looks not very pleasant and leads that way.  Very interesting, we end up having to use die_on_timeout => 0 because of API change. I do not think this is what this API change initially would like to see.  
Maybe you use type_string here instead of script_run ? So we can avoid change to other tests ?

Reply:
Same the above comment. we'd better to keep with die_on_timeout =>1 for non-s390x.

Reply:
> @guoxuguang I think you'd better raise this issue to openQA project or let them know. It seems that the change does not take this usage on s390x into consideration which means you have to use die_on_timeout => 0. Now you made the change which branches off s390x. I did not say branching off here is really bad here, but it does force us to make the same change to other arches here to some extent because it looks not very pleasant and leads that way. Very interesting, we end up having to use die_on_timeout => 0 because of API change. I do not think this is what this API change initially would like to see. API option should be optional to use instead have to use. Maybe you use type_string or other script_xxxx here instead of script_run ? So we can avoid change to other tests ?

Sure, great idea. I need to create a ticket to openQA tools team to let them know that maybe this new change does not take care this usage on s390x. 
Also, I had confirmed that `type_string()` still does no work for svirt backend. So far, just  `virt_utils::lpar_cmd()` and `script_run() `work with svirt backend.  yes, I will try to confirm the other API work with svirt backend. thanks so much for your great help and suggestion.